content/hardware/04.pro/boards/portenta-x8/tutorials/datalogging-iot/content.md

```diff
@@ -0,0 +1,416 @@
+
+How can we plan, build and deploy IoT solutions? To answer that question, we must think about **IoT architecture**. Due to the different IoT devices and applications that exist and can exist, there is not just one unique architecture for IoT devices and applications. But, we can talk about a base architecture that can be considered as a starting point for every IoT project. This base architecture consists of **three essential layers**: **perception** (or devices), **network**, and **application**. Let's talk more about these layers:
+
+- **Perception layer**: this is the sensor's layer, where data comes from. In this layer, data is gathered with one or more sensor nodes; actuators, that answer to data collected from sensor nodes, are also in this layer.
+- **Network layer**: this is the layer where data from sensor nodes is recollected and then transmitted to back-end services, such as databases. 
+- **Application layer**: this layer is what the device or application user sees and interacts with, for example, a dashboard. 
+
+The three-layer IoT architecture can be a starting point for designing and implementing an IoT device or application. In this tutorial, we are going to take this base architecture and set up a data logging application, as shown in the image below:
+
+![IoT application high-level architecture.](assets/x8-data-logging-img_01.png)
+
+In the high-level architecture described in the image above:
+
+- The perception layer consists of an MKR WiFi 1010 board; this board will gather information from a sensor. 
+- The network layer consists of the MQTT broker (Mosquitto), the data forwarder (Node-RED), and the database (InfluxDB). 
+- The application layer consists of a dashboard (Grafana) where information from the sensor node is shown.
+
+***For documentation purposes, we are going to explain, step by step, the installation process of each part of the application (Mosquitto, Node-RED, InfluxDB, and Grafana); this process can be done quickly using a unique Compose `YAML` file. If you are familiar with Containers in Docker, you can skip to this section of the tutorial.***
+
+Let's start by configuring the MQTT broker!
+
+## Installing Mosquitto
+
+Let's start by creating a new directory in our Portenta X8 called `mqtt`; inside this directory, we are going to make a file named `docker-compose.yml`: 
+
+```
+# mkdir mqtt
+# cd mqtt
+# export TERM=xterm
+# stty rows 36 cols 150
+# sudo vi docker-compose.yml
+```

```

Comment:
This is personal, but the `#` is not super clear to me in some of the commands that you show in the tutorial

Maybe it is caused by the lack of CLI formatting that we have right now.

Reply:
Yeah, for me too is not super clear. The # in that section refers to the bash of the MQTT broker, I didn't know how to format it better...

Reply:
What about this linux format, and using ` ```bash `
```bash
~$ cd folder
~/folder$ echo Message
Message
```

Reply:
Or windows style with ` ```cmd `
```cmd
C:\> cd folder
C:\folder\> echo Message
Message
```

Reply:
I think it is reasonable if this is the command prompt the user will see on their machine.

However, I'm not sure that is the case. The standard prompt is `$`. `#` indicates the user is `root` (su). You can see how it changes in the second screenshot here:

https://docs.arduino.cc/tutorials/portenta-x8/out-of-the-box#ssh

`su` should only be used when really necessary. I'm not at all knowledgeable about this specific application, but from a quick look I don't see any reason why it would be necessary here. The fact that `sudo` is used in one of the commands indicates that it is not necessary (it is more common to use `sudo` to run a specific command as superuser when that is required, running all the other commands as a normal user.

For this reason, the reader may well have never seen this special prompt, whereas they are much more likely to be familiar with the standard `$`.

`#` is also problematic because this is the shell comment character. When I see `#` in this context, my first impression is that it is a comment.

Reply:
@canchebagur What do you think, maybe switching the `#` to `$` to align with the rest of tutorials for now, until we find a better solution? 🤔

Reply:
>What about this linux format, and using ```bash

In the example you showed, it looks nice, but I often find that this produces inappropriate syntax highlighting when used on blocks containing command line invocations, so I avoid it. It is definitely the right way to go when the block contains an actual Bash script though.

Here is an example of what I mean by inappropriate syntax highlighting:

https://arduino.github.io/arduino-cli/0.26/getting-started/#compile-and-upload-the-sketch

Reply:
I would change the # for $ to be aligned with the rest of the tutorials and start researching a better solution for formatting this. We are going to need a better solution for the upcoming Portenta X8 tutorials :)

Reply:
totally agree, sounds good to me then 😄